app/views/NotFoundPage.scala.html

```diff
@@ -0,0 +1,27 @@
+@*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import components.Layout
+
+@this(layout: Layout)
+
+@()(implicit request: MessagesRequest[?], messages: Messages)
```

Comment:
I think this could be `implicit request: RequestHeader` and leave out changing the given request to MessageRequest - because layout only wants a request header and then we are explicitly separately passing messages (where a messages request wraps messages)

rather than via that wrapped way we are using I18nSupport from play to get implicit conversion of request to messages https://www.playframework.com/documentation/3.0.x/api/scala/play/api/i18n/I18nSupport.html